plugins/modules/vyos_prefix_lists.py

```diff
@@ -0,0 +1,133 @@
+
+"""
+The module file for vyos_prefix_lists
+"""
+
+from __future__ import absolute_import, division, print_function
+
+__metaclass__ = type
+
+DOCUMENTATION = """
+module: vyos_prefix_lists
+short_description: Prefix-Lists resource module for VyOS
+description:
+  - This module manages prefix-lists configuration on devices running VyOS
+version_added: 2.3.1
+author: Priyam Sahoo (@priyamsahoo)
+notes:
+  - Tested against VyOS 1.1.8 (helium)
+  - This module works with connection C(network_cli)
+options:
+  config:
+    description: A list of prefix-list options
+    type: list
+    elements: dict
+    suboptions:
+      afi:
+        description: The Address Family Indicator (AFI) for the prefix-lists
+        type: str
+        choices: ["ipv4", "ipv6"]
+        required: true
+      prefix_lists:
+        description: A list of prefix-list configurations
+        type: list
+        elements: dict
+        suboptions:
+          name:
+            description: The name of a defined prefix-list
+            type: str
+            required: true
+          description:
+            description: A brief text description for the prefix-list
+            type: str
+          rules:
+            description: Rule configurations for the prefix-list
+            type: list
+            elements: dict
+            suboptions:
+              id:
+                description: A numeric identifier for the rule
+                type: int
+                required: true
+              description:
+                description: A brief text description for the prefix list rule
+                type: str
+              action:
+                description: The action to be taken for packets matching a prefix list rule
+                type: str
+                choices: ["permit", "deny"]
+              ge:
+                description: Minimum prefix length to be matched
+                type: int
+              le:
+                description: Maximum prefix list length to be matched
+                type: int
+              prefix:
+                description: IPv4 or IPv6 prefix in A.B.C.D/LEN or A:B::C:D/LEN format
+                type: str
+  running_config:
+    description:
+      - This option is used only with state I(parsed).
+      - The value of this option should be the output received from the VyOS device
+        by executing the command B(show configuration commands | grep prefix-list).
+      - The state I(parsed) reads the configuration from C(running_config) option and
+        transforms it into Ansible structured data as per the resource module's argspec
+        and the value is then returned in the I(parsed) key within the result.
+    type: str
+  state:
+    description:
+      - The state the configuration should be left in
+    type: str
+    choices:
+      - merged
+      - replaced
+      - overridden
+      - deleted
+      - gathered
+      - rendered
+      - parsed
+    default: merged
+"""
+EXAMPLES = """
```

Comment:
Please add examples for all the state operations. If you need help you can refer  https://github.com/ansible-collections/vyos.vyos/blob/8264a29424c79c25d03cf89c376a72ec6a0f600f/plugins/modules/vyos_route_maps.py#L262